### PR TITLE
OpenZipkin/Cassandra StorageType to turn On/off Hostname verification #3427

### PR DESCRIPTION
**Pull request by priyavivek2307**

### CODE REVIEW

File: zipkin-server/src/main/resources/zipkin-server-shared.yml [hunk 1]
```diff
@@ -144,6 +146,8 @@ zipkin:
       index-fetch-multiplier: ${CASSANDRA_INDEX_FETCH_MULTIPLIER:3}
       # Using ssl for connection, rely on Keystore
       use-ssl: ${CASSANDRA_USE_SSL:false}
+      # Override hostname verification in Cassandra setting
+      override-hostname-verification: ${CASSANDRA_OVERRIDE_HOSTNAME:false}
```
Comment (jcchavezs) [inline zipkin-server/src/main/resources/zipkin-server-shared.yml]: I'd rather call the var `CASSANDRA_OVERRIDE_HOSTNAME_VERIFICATION` to be consistent.

Comment (priyavivek2307) [reply zipkin-server/src/main/resources/zipkin-server-shared.yml]: updated the variable name as suggested. @jcchavezs

File: zipkin-storage/cassandra/src/main/java/zipkin2/storage/cassandra/internal/SessionBuilder.java [hunk 2]
```diff
@@ -77,7 +79,12 @@ public static CqlSession buildSession(
     config = config.withBoolean(REQUEST_DEFAULT_IDEMPOTENCE, true);
 
     if (useSsl) config = config.withClass(SSL_ENGINE_FACTORY_CLASS, DefaultSslEngineFactory.class);
-
+    
+    if (overrideHostnameVerification)
```
Comment (ankit-gautam23) [inline zipkin-storage/cassandra/src/main/java/zipkin2/storage/cassandra/internal/SessionBuilder.java]: I think we can write this in one line, excluding the need of if-else.

Replacing this

if (overrideHostnameVerification) 
    	config = config.withBoolean(SSL_HOSTNAME_VALIDATION, true);
    else
    	config = config.withBoolean(SSL_HOSTNAME_VALIDATION, false);

With this

config = config.withBoolean(SSL_HOSTNAME_VALIDATION, overrideHostnameVerification);

**Review by ankit-gautam23 — CHANGES_REQUESTED** [review]

**Review by codefromthecrypt — COMMENTED** [review]
if you can use the square formatter, the change will be simpler as it won't change so many lines. I think you are using tabs where the project uses spaces

**Comment by codefromthecrypt** [comment]
I think to get this across the finish line, it might be faster to redo it. Reason is that the cassandra property is `SSL_HOSTNAME_VALIDATION` and so that means our property should be `CASSANDRA_SSL_HOSTNAME_VALIDATION` for env and sslHostnameValidation for the field. using different names can cause confusion. Most of the work not done here is adding the spring yaml plumbing to propagate this setting, as well update the zipkin-server README section for cassandra.

File: zipkin-server/src/main/resources/zipkin-server-shared.yml (hunk 1 above)
Comment (codefromthecrypt) [reply zipkin-server/src/main/resources/zipkin-server-shared.yml]: for future, any time we are going to add a new variable, follow existing conventions first. The easiest way and regardless of cassandra or not, is to do a web or github search for the property and use that. If there are multiple styles used, prefer the most used, for example other tools that configure cassandra.

Regardless, we don't want any incoherence in our codebase, so we needn't invent anything and just reuse the norm. My follow-up PR does that, just explaining here as I noticed sometimes folks accidentally don't look and then ask others to use non-conventional names. https://github.com/apache/cassandra-java-driver/blob/8d5849cb38995b312f29314d18256c0c3e94cf07/core/src/main/java/com/datastax/oss/driver/api/core/config/TypedDriverOption.java#L230

**Comment by codefromthecrypt** [comment]
reopened in https://github.com/openzipkin/zipkin/pull/3701 as I was unable to alter the branch under this PR.

**Comment by codefromthecrypt** [comment]
https://github.com/openzipkin/zipkin/releases/tag/3.0.5 includes this

**Pull request closed without merging**